2022/incoming/sleigh-odometer.pod

```diff
@@ -0,0 +1,106 @@
+Author: Thibault DUPONCHELLE <thibault.duponchelle@gmail.com>
+Title: The Sleigh odometer
+Topic: state
+
+=encoding utf8
+
+=head1 Sleigh odometer
+
+=head2 Customize the sleigh
+
+It's already December and elves are working hard to produce all toys for Christmas.
+
+On the other side, Santa is preparing his Sleigh, cleaning or customizing.
+ 
+This year, in addition to a temperature sensor, Santa asked to install an odometer to his sleigh so that he can count each delivery.
+For the record, the distance unit is not kilometers or miles nor inches but "elf little toe"... but it does not change what follows :)
+
+In order to implement the counter of the odometer, an elf came with this code:
+
+=begin perl
+
+#!/usr/bin/env perl
+use 5.010;
+
+sub count() {
+        my $distance = 0;
+        return ++$distance;
+}
+
+count(); count(); 
+
+say count();
+
+=end perl
+
+But sadly it does not work as expected because the distance is reset at each C<count()> call :/
+
+=head2 Asking experienced elves
+
+After asking some more experienced elf, the elf modified to implement a persistent variable via a "lexical trick":
+
+=begin perl
+
+#!/usr/bin/env perl 
+use 5.010;
+
+{
+    my $distance = 0;
+    sub count {
+        return ++$distance;
+    }
+}
+
+count(); count(); count(); count();
+say count();
+
+=end perl
+
+It works!
+
+Some other experienced elf proposed to use a "closure" that he perceived as "more elegant": 
+
+=begin perl
+
+#!/usr/bin/env perl 
+use 5.010;
+
+sub make_counter() {
+    my $distance = 0;
+    return sub {
+        return ++$distance;
+    }
+}
+
+my $counter = make_counter();
+
+$counter->(); $counter->(); $counter->(); $counter->();
+say $counter->();
+
+=end perl
+
+=head2 Final word from Santa
+
+These 2 previous code works well, butt there is definitely a simpler way! 
```

Comment:
Should probably be "pieces of code" or "code snippets"?

Also very unfortunate typo: "butt" ;)

Reply:
Good point thank you